protocol/storage/client.go

```diff
@@ -251,11 +254,10 @@ func (smc *Client) checkDealResponse(ctx context.Context, resp *storagedeal.Resp
 }
 
 func (smc *Client) minerForProposal(c cid.Cid) (address.Address, error) {
-	storageDeal := smc.api.DealGet(c)
-	if storageDeal == nil {
-		return address.Undef, fmt.Errorf("no such proposal by cid: %s", c)
+	storageDeal, err := smc.api.DealGet(c)
+	if err != nil {
+		return address.Undef, errors.Wrapf(err, "no such proposal by cid: %s", c)
```

Comment:
This message could now be misleading, depending on the error.  I suggest tweaking the wording to "failed to fetch deal".